site/profile/manifests/users.pp

```diff
@@ -171,4 +172,19 @@
     line    => "${name} ALL=(ALL) NOPASSWD:ALL",
     require => File['/etc/sudoers.d/90-puppet-users'],
   }
+
+  if $authenticationmethods != '' {
+    file { "/etc/ssh/sshd_config.d/authenticationmethods_${name}.conf":
+      ensure  => file,
+      mode    => '0644',
+      owner   => 'root',
+      group   => 'root',
+      notify  => Service['sshd'],
+      content => @("EOT")
+Match User ${name}
+  AuthenticationMethods ${authenticationmethods}
+Match all
```

Comment:
If there is more than one local user that define a value for `authenticationmethods`, this will result in `Match all` being defined more than once in sshd configuration. The `Match all` statement should be added to `/etc/sshd/sshd_config` if it is necessary.

Reply:
From what I read, the `Match all` is essentially the "end block" statement of a `Match User xxxxx` statement. If it is not there, it means that whatever content comes after that file in whatever order the files are included will only apply to `${name}`, so I think the `Match all` clause must be there.

Reply:
Using `sshd_config` resource from [augeasprovider_ssh](https://forge.puppet.com/modules/puppet/augeasproviders_ssh) simplifies what needs to be done:
```
  sshd_config { "${name} authenticationmethods":
    ensure      => present,
    condition   => "User ${name}",
    key         => 'authenticationmethods',
    value       => ${authenticationmethods},
  }
```

Augeas will take care of putting the Match blocks at the right place in the config file. No need to have one file per user, no need to handle the creation of `sshd_config.d`.

Reply:
fixed for using augeas. Should I actually remove the ensure of sshd_config.d ?

Reply:
Yes, remove everything related to `sshd_config.d`.

Reply:
done